treeherder/webapp/api/jobs.py

```diff
@@ -147,6 +147,20 @@ def retrigger(self, request, project, jm, pk=None):
         else:
             return Response("No job with id: {0}".format(pk), 404)
 
+    @action(permission_classes=[IsAuthenticated])
```

Comment:
This permission class only check if a user is authenticated. As far as I understand you wanted to expose this functionality only to sheriffs, right? In order to do do that, you need to add a new permission class to treeherder/webapp/api/permissions.py, something like this:
```python
class IsStaff(permissions.BasePermission):

    """
    The request is authenticated as an admin staff (eg. sheriffs).
    """

    def has_permission(self, request, view):
        return (request.user and request.user.is_authenticated() and
                   request.user.is_staff)
```
and use that class to the action decorator